src/analysis/rule_extraction.jl

```diff
@@ -0,0 +1,88 @@
+using SIRUS, MLJ
+using StableRNGs: StableRNG
+import ..performance: temporal_variability
+
+"""
+    Rule{V<:Vector{Float64},W<:Vector{Vector{Any}}}
+
+# A Rule contains a condition (vector of conditional clauses) and consequent (vector 
+of values for then and else results of the condition)
+"""
+struct Rule{V<:Vector{Float64},W<:Vector{Vector{Any}}}
+    condition::W
+    consequent::V
+end
+
+"""
+    rules(rules::SIRUS.StableRules)
+
+Vector of Rule objects. These are **not** the same as SIRUS.Rule object.
+
+# Arguments
+- `rules` - SIRUS.StableRules object
+
+# Return
+Vector of Rule objects
+"""
+function rules(rules::SIRUS.StableRules{Float64})
+    [Rule(_condition(rules, i), _consequent(rules, i)) for i in eachindex(rules.rules)]
+end
+
+"""
+    _condition(rules::SIRUS.StableRules{Float64}, index::Int64)
+
+Vector containing condition clauses. Each condition clause is a vector with three 
+components: a feature_name::String, a direction::String (< or ≤) and a value:<Float64 
+
+# Arguments
+- rules - SIRUS.StableRules object containing all rules
+- `index` - Index of the rule
+
+# Return
+Vector of vectors of rule conditions.
+"""
+function _condition(rules::SIRUS.StableRules{Float64}, index::Int64)
+    feature_name(split) = split.splitpoint.feature_name
+    direction(split) = split.direction
+    value(split) = split.splitpoint.value
+
+    [[feature_name(s), direction(s), value(s)] for s in rules.rules[index].path.splits]
+end
+
+"""
+    _consequent(rules::SIRUS.StableRules{Float64}, index::Int64)
+
+Vector of Rule consequent with two components: the probability of the 'then' clause and 
+probability of the 'else' clause (here called otherwise)
+
+# Arguments
+- rules - SIRUS.StableRules object containing all rules
+- `index` - Index of the rule
+
+# Return
+Vector of rule consequents
+"""
+function _consequent(rules::SIRUS.StableRules{Float64}, index::Int64)
+    weight = rules.weights[index]
+    rule = rules.rules[index]
+    then_probability = SIRUS._simplify_binary_probabilities(weight, rule.then)
+    otherwise_probability = SIRUS._simplify_binary_probabilities(weight, rule.otherwise)
```

Comment:
Relying on another package's private functions 😬 

Okay if there's no other way, but it does make me feel uncomfortable

Reply:
https://github.com/rikhuijzer/SIRUS.jl/blob/78d74aa1cebfaf564fa6e547d03e0f8a3f96a7d4/src/ruleshow.jl#L31

This is where this function is being defined. The only other way around this I see is to define the same function inside ADRIA... which is not great too. What do you think?

Reply:
Agree that neither option is great.

Redefining the function inside ADRIA works so long as other aspects of SIRUS don't change.
Relying on SIRUS' private functions will only work so long as nothing in these functions changes (especially returned types).

I think leave it as is for now. Future Pedro can fix if anything breaks 😛